### ISSUE TITLE
Metaclass conflict - can't run Rekall

### ISSUE THREAD

**Issue by querist** [google/rekall#116]
Please see attached for error message.
I just cloned the repository and ran setup.py install to build and install rekall on my personal windows machine. This worked perfectly on my work windows 8.1 system, but on my personal windows 8.1 running rekall generates a "metaclass conflict" message as seen in the attached file.

Thanks
[error.txt](https://github.com/google/rekall/files/120489/error.txt)

**Comment by scudette** [comment]
Hi, 
  This is a very strange error. Can you please indicate which version of yaml you are using?

```
In [3]: import yaml

In [4]: print yaml.__version__
3.11
```

Also it does not look like you are running in a virtual env. The best way to run is inside a virtual env to make sure all versions are exactly right. Can you please try again inside a virtualenv? (Just follow the instructions in http://rekall-forensic.blogspot.ch/2015/09/installing-rekall-on-windows.html)

**Comment by querist** [comment]
I tried a virtual env and had the exact same error. 

I'm running version 3.11 of yaml.

**Comment by scudette** [comment]
Could you please try to reverse the order of the inheritance?

```
class OrderedYamlDict(yaml.YAMLObject, collections.OrderedDict):
```
Change to 
```
class OrderedYamlDict(collections.OrderedDict, yaml.YAMLObject):
```

**Comment by querist** [comment]
Same error.

2016年2月6日星期六，Michael Cohen <notifications@github.com> 写道：

> Could you please try to reverse the order of the inheritance?
>
> class OrderedYamlDict(yaml.YAMLObject, collections.OrderedDict):
>
> Change to
>
> class OrderedYamlDict(collections.OrderedDict, yaml.YAMLObject):
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/rekall/issues/116#issuecomment-180890816>.
>


-- 
----------------------------
妇女能顶半边天
-- 毛泽东

**Comment by scudette** [comment]
Thanks for helping debug this. Its a big hard because I can not reproduce. If you could please add the following code to that file:

```
print yaml.YAMLObject.__metaclass__
try:
   print collections.OrderedDict.__metaclass__
except:
   print "No metaclass"
```

Apparently this error occurs when both classes have incompatible metaclasses but (at least on my system) the collections.OrderedDict object has no metaclass at all.

Can you also please indicate which version of python are you running. Just paste the banner from the top of the python command. For example mine is:

```
Python 2.7.6 (default, Jun 22 2015, 17:58:13) 
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
```

**Comment by querist** [comment]
That solved the problem - I upgraded from 2.7.1 to 2.7.10 and it works now. I did not need to rebuild or anything. This is strange.
Thanks!
